packages/lib/server-only/user/update-password.ts

```diff
@@ -3,50 +3,71 @@ import { compare, hash } from 'bcrypt';
 import { prisma } from '@documenso/prisma';
 
 import { SALT_ROUNDS } from '../../constants/auth';
+import { isPasswordNull } from './is-password-null';
 
 export type UpdatePasswordOptions = {
   userId: number;
   password: string;
-  currentPassword: string;
+  currentPassword?: string;
 };
 
 export const updatePassword = async ({
   userId,
   password,
   currentPassword,
 }: UpdatePasswordOptions) => {
+  console.log('userId', userId, 'update passowrd');
   // Existence check
   const user = await prisma.user.findFirstOrThrow({
     where: {
       id: userId,
     },
   });
 
-  if (!user.password) {
+  if (!user) {
     throw new Error('User has no password');
   }
 
-  const isCurrentPasswordValid = await compare(currentPassword, user.password);
-  if (!isCurrentPasswordValid) {
-    throw new Error('Current password is incorrect.');
-  }
+  const isUserPasswordNull = await isPasswordNull({ id: userId });
 
-  // Compare the new password with the old password
-  const isSamePassword = await compare(password, user.password);
-  if (isSamePassword) {
-    throw new Error('Your new password cannot be the same as your old password.');
-  }
+  if (isUserPasswordNull) {
+    const hashedNewPassword = await hash(password, SALT_ROUNDS);
 
-  const hashedNewPassword = await hash(password, SALT_ROUNDS);
+    const updatedUser = await prisma.user.update({
+      where: {
+        id: userId,
+      },
+      data: {
+        password: hashedNewPassword,
+      },
+    });
 
-  const updatedUser = await prisma.user.update({
-    where: {
-      id: userId,
-    },
-    data: {
-      password: hashedNewPassword,
-    },
-  });
+    return updatedUser;
+  } else {
+    if (currentPassword && user.password) {
+      const isCurrentPasswordValid = await compare(currentPassword, user.password);
+      if (!isCurrentPasswordValid) {
+        throw new Error('Current password is incorrect.');
+      }
+
+      // Compare the new password with the old password
+      const isSamePassword = await compare(password, user.password);
+      if (isSamePassword) {
+        throw new Error('Your new password cannot be the same as your old password.');
+      }
+
+      const hashedNewPassword = await hash(password, SALT_ROUNDS);
 
-  return updatedUser;
+      const updatedUser = await prisma.user.update({
+        where: {
+          id: userId,
+        },
+        data: {
+          password: hashedNewPassword,
+        },
```

Comment:
This should be simplified to only need an `if` branch if the password is not null where we can check and throw otherwise we proceed with the update. 

Right now it's doing the work twice and is a bit more confusing since `else` branches become hard to follow.